### DISCUSSION TITLE
Features to Add/Improvements (Running List)

### DISCUSSION THREAD

**Discussion by coreywillwhat** [audiconnect/audi_connect_ha#323] [Ideas]
### Improvements
- `Coordinator` to allow official front end changes to polling and spook integration
- 15 min Refresh interval works but might be risky depending on actions and restart. Default should be higher with warning if 15 is selected
- Use `RepairsFlow` to initiate restart after options update
- Add User Option for `UPDATE_SLEEP` > Advanced Option
- Make some user options under the configure menu only for 'advanced' mode
    - Update at Startup
    - UPDATE_SLEEP
- Get vehicle health warnings (Sensor)
    - captured one in #207 

### Clean-Up
- Better Logging
- Define defaults for `start_climate_control` within `audi_services.py` rather than `audi_account.py` to better reflect user selections in logging and cleaner code review.
- readme

### Completed
- Built in delay between vehicle action and cloud update

**Comment by cdnninja** [comment]
Totally agree.  A part of this is breaking out the api as well. I referenced the architecture over at kia_uvo as a reference point.   I find it much easier to manage and understand.

**Reply by cdnninja** [reply]
https://github.com/Hyundai-Kia-Connect/ has one repo for the api one for the home assistant side.

**Reply by coreywillwhat** [reply]
That was done over on cyr-ius's fork. Agree that would be a nice path to go wit this.

**Reply by cdnninja** [reply]
Yeah we chatted on it here: https://github.com/audiconnect/audi_connect_ha/discussions/273  my hope was to have him join us, potentially using that API has the source of moving this forward however I haven't heard back from him.     Could be busy with life. :) 

My goal is to have this repo as an inclusive collaboration so not dependent on any one person. Since really we all have lives and other things that come up.

**Reply by cdnninja** [reply]
I took the liberty of creating a repo: https://github.com/audiconnect/audi_connect_api

I will start framing up the plumbing. Feel free to make commits to it anytime you like.

**Reply by cdnninja** [reply]
@cyr-ius would you be interested in collaborating?

**Reply by coreywillwhat** [reply]
I've never done any home assistant dev so its been a steep learning curve. I'm sure @cyr-ius could give us some good insight. Put me out of a job haha

**Comment by jeffkibuule** [comment]
Is it possible to clean up the data coming back from the API. Right now, it looks like it's a passthrough in all lower case, which isn't ideal. Same with the name of some of the sensors. I'd be happy to figure out how to do a PR for this.

**Comment by FrankC-BEL** [comment]
Would it be possible to also get the vehicle location via the API?  This would be great to trigger some automations when the car is home.

**Comment by MarcW22** [comment]
Hi, first of all thank you guys very much for all the work and effort you put into the this integration. I really appreciate it! But nevertheless i would like to ask if there are any chances to implement sending targets for the navigation system?

Thanks in advance

**Reply by coreywillwhat** [reply]
Thanks for the suggestion! Is it possible to use the myAudi app for this? I only have the option to navigate via pedestrian route and my phone. I can’t send addresses to the vehicle navigation system. If you have the option, could you please send me screenshots?

**Reply by MarcW22** [reply]
Thanks for the fast answer!
Yes it's possible and works fine with the my audi app.
Attached the screenshot. ![image](https://github.com/user-attachments/assets/0a4fc899-b9d5-4b5f-add6-e82ada788b1f)

**Reply by coreywillwhat** [reply]
I've opened #548 as a feature request. I need to get the primary services up and running, then will be working on enhancements such as this. Thanks for raising this!

**Reply by MarcW22** [reply]
Thanks a lot. Let me know in case i can help with anything. Unfortunately I can't code but testing etc. is possible.

**Comment by coreywillwhat** [comment]
Closing this discussion in favor of a new discussion #554 which is a better place to discuss the direction we are going.
